cli/internal/otel/receiver.go

```diff
@@ -46,15 +46,17 @@ type Consumer struct {
 	metricsFilename string
 	metricsFile     *os.File
 	consumeMetric   func(context.Context, pluginMetric)
+	wg              *sync.WaitGroup
 }
 
 func (c *Consumer) Shutdown(ctx context.Context) {
 	if c.quit != nil {
 		close(c.quit)
+		c.wg.Wait()
```

Comment:
Shouldn't we wait regardless of the quit channel being set or not?

Reply:
I think we can remove the `nil` check it will never be `nil` per https://github.com/cloudquery/cloudquery/blob/92c81752ddc95617c48326df97aea353400525fa/cli/internal/otel/receiver.go#L233

Reply:
Done in https://github.com/cloudquery/cloudquery/pull/19409/commits/c168f1d8a165f773546e7df29fb95b1c671407b3